### DISCUSSION TITLE
Help understanding elemental variables

### DISCUSSION THREAD

**Discussion by aprilnovak** [idaholab/moose#17273] [Q&A General] [answered]
Hi all,

I'm trying to get a better understanding of the `MultiAppNearestNodeTransfer`, but am finding myself confused on the interpretation of elemental variables. On the MOOSE website, it says:

> for elemental shape functions the DOF values are not associated with nodes.

but rather with quadrature points. But then in the transfer itself, the target points in the receiving app for FIRST MONOMIAL are set to the nodes of element: ([code here](https://github.com/idaholab/moose/blob/next/framework/src/transfers/MultiAppNearestNodeTransfer.C#LC235:~:text=%2F%2F%20For%20L2_LAGRANGE%2C%20we%20take%20all,%7D))

```
if (elemental) {

           // For L2_LAGRANGE, we take all the nodes of element
            for (auto & node : elem->node_ref_range())
            {
              points.push_back(node);
              point_ids.push_back(node.id());
            }

}
```

How do these two statements reconcile with one another? I'm trying to figure out why transfers to/from FIRST MONOMIAL give strange-looking results for this transfer ([see here](https://github.com/idaholab/moose/issues/17246)), and I feel like this (apparent, to me) inconsistency might explain something.

Thanks!

**Comment by loganharbour** [comment, marked as answer]
In libMesh, the dofs for elemental variables are represented on the element themselves.

Let's say you have a system with just a single first order, elemental variable and QUAD4 elems. Each QUAD4 elem would have 4 dofs associated with it (a value at each "node"; note I use "node" here because it's not actually stored on the Node object!). Then, all of the physical Node objects (which represent the vertices for each QUAD4 elem, which are shared between elems) have no dofs whatsoever. In addition, for first-order L2_LAGRANGE variables, the spatial location of each node (where the dofs live physically in space) happen to also be the physical location of the Nodes - even though the Node objects do not own the dofs!

Within the loop that you linked, we're specifying that we need a value for each of the dof values for the L2_LAGRANGE element (again, the Node locations _are_ the locations for the dofs, even though the dofs don't _live_ on the Node objects). I'm still working through the validity of using the Node ID (stored [here](https://github.com/idaholab/moose/blob/3a4c81fd417de334a120fea45273c7a394b2adbd/framework/src/transfers/MultiAppNearestNodeTransfer.C#L603)) for the purposes of mapping the incoming evaluations [here](https://github.com/idaholab/moose/blob/3a4c81fd417de334a120fea45273c7a394b2adbd/framework/src/transfers/MultiAppNearestNodeTransfer.C#L625).
